### PR TITLE
Use DTZ rule and fix code that breaks it

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Small problem found: we have no linter verifying that we use timezones in consistent manner.

- Adding "DTZ" rule for Ruff: https://beta.ruff.rs/docs/rules/#flake8-datetimez-dtz
- Fixing code that breaks the rules

### CODE REVIEW

**Review by seanstory — DISMISSED** [review]
🥇

**Comment by artem-shelkovnikov** [comment]
@seanstory I had to fix more things 😅 Can I get another stamp on this PR?

File: .ruff.toml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-select = ["E", "F", "B"]
+select = ["E", "F", "B", "DTZ"]
```
Comment (tarekziade) [inline .ruff.toml:1]: yay

**Review by tarekziade — DISMISSED** [review]

**Review by seanstory — DISMISSED** [review]

File: connectors/services/job_scheduling.py [hunk 2]
```diff
@@ -200,6 +200,9 @@ async def _should_schedule_scheduled_sync(job_type):
                 )
                 return False
 
+            print("NEXT is {next_sync}")
+            print("NOW is {now}")
```
Comment (wangch079) [inline connectors/services/job_scheduling.py:204]: debug log?

File: connectors/services/job_scheduling.py [hunk 3]
```diff
@@ -176,7 +176,7 @@ async def _should_schedule_scheduled_sync(job_type):
                 return False
 
             job_type_value = job_type.value
-            now = datetime.utcnow()
+            now = datetime.now(timezone.utc)
```
Comment (wangch079) [inline connectors/services/job_scheduling.py:179]: Should we move all the datetime related functions to `utils` or a new `datetime` module?

File: connectors/services/job_scheduling.py (hunk 2 above)
Comment (artem-shelkovnikov) [reply connectors/services/job_scheduling.py:204]: Yup, good catch!

File: connectors/services/job_scheduling.py (hunk 3 above)
Comment (artem-shelkovnikov) [reply connectors/services/job_scheduling.py:179]: Eventually yes, should be something like `utils/datetime.py`. I'm not planning it in this PR though, I already have a lot of problems with this change alone.

**Comment by artem-shelkovnikov** [comment]
Closing to re-open later, when implemented properly

**Pull request closed without merging**